aidb/inference/bound_inference_service.py

```diff
@@ -169,24 +170,32 @@ async def _insert_output_results_in_tables(self, output_results_data: List[pd.Da
         await conn.run_sync(lambda conn: tmp_df.to_sql(table, conn, if_exists='append', index=False))
 
 
+  async def _check_inputs_in_cache_table(self, inputs: pd.DataFrame, conn):
+    """
+    checks the presence of inputs in the cache table
+    """
+    cache_entries = await conn.run_sync(lambda conn: pd.read_sql(text(str(self._cache_query_stub.compile())), conn))
```

Comment:
Am I understanding that this fetches the entire cache table at once?

Reply:
Yes. It keeps them in the memory.

Reply:
Is there a way to take into account the inputs? Seems a bit heavy for large tables

Reply:
I don't have another good idea. If you think, this can be a problem, we may not merge it at this time?

Reply:
Raise an issue please. I think we can have a big IN clause or think of something else creative

Reply:
#111